### PR TITLE
WT-8580 Fix the search in history store insert to check key order

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/cursor/cur_hs.c [hunk 1]
```diff
@@ -938,10 +939,11 @@ __curhs_insert(WT_CURSOR *cursor)
...
 #ifdef HAVE_DIAGNOSTIC
     /* Do a search again and call next to check the key order. */
-    WT_WITH_PAGE_INDEX(session, ret = __curhs_search(cbt, false));
-    WT_ASSERT(session, ret == 0);
-    if (cbt->compare == 0)
+    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = file_cursor->search(file_cursor));
+    if (ret == 0)
         WT_ERR_NOTFOUND_OK(__curhs_file_cursor_next(session, file_cursor), false);
+    else
+        WT_ASSERT(session, ret == WT_NOTFOUND);
```
Comment (kommiharibabu) [inline src/cursor/cur_hs.c:946]: Why do you expect the search can return WT_NOTFOUND? The update is just inserted.

Comment (quchenhao) [reply src/cursor/cur_hs.c:946]: We the search the initial page can get lost. Eviction can kick in and delete that key if it is already obsolete.

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/cursor/cur_hs.c [hunk 2]
```diff
@@ -938,10 +939,9 @@ __curhs_insert(WT_CURSOR *cursor)
 
 #ifdef HAVE_DIAGNOSTIC
     /* Do a search again and call next to check the key order. */
-    WT_WITH_PAGE_INDEX(session, ret = __curhs_search(cbt, false));
+    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = file_cursor->search(file_cursor));
```
Comment (agorrod) [inline src/cursor/cur_hs.c:942]: Why do some calls to __curhs_search use a WT_WITH_PAGE_INDEX macro and others don't? It seems like something that should be completely consistent.

Comment (quchenhao) [reply src/cursor/cur_hs.c:942]: @agorrod  I checked all the usage of __curhs_search. They are all wrapped with WT_WITH_PAGE_INDEX. Can you point me to which usage?

Comment (quchenhao) [reply src/cursor/cur_hs.c:942]: I checked that all the usage of __curhs_search are wrapped with WT_WITH_PAGE_INDEX. I think you are confused with some function calls that prefix with "__curhs_search", such as "__curhs_search_near_helper"

[the remark this answers is not in the document]
Comment (quchenhao) [reply src/cursor/cur_hs.c]: Fixed in [9acdac3](https://github.com/wiredtiger/wiredtiger/pull/7751/commits/9acdac3ed06ecd7dd5563b139b3196bed1db3692)

**Pull request merged**